cloudwatch_metrics_collector/src/metrics_collector.cpp

```diff
@@ -101,10 +115,51 @@ int MetricsCollector::RecordMetrics(
     AWS_LOGSTREAM_DEBUG(__func__, "Recording metric with name=[" << metric_msg->metric_name << "]");
 
     // create a MetricObject with message parameters to batch
-    Aws::CloudWatchMetrics::Utils::MetricObject metric_object {metric_msg->metric_name,
+    Aws::CloudWatchMetrics::Utils::MetricObject metric_object{metric_msg->metric_name,
                                                               metric_msg->value,
                                                               metric_msg->unit,
-                                                              GetMetricDataEpochMillis(*metric_msg),
+                                                              GetMetricDataEpochMillis(metric_msg->time_stamp),
+                                                              dimensions,
+                                                              this->storage_resolution_.load()};
+    bool batched = metric_service_->batchData(metric_object);
+
+    if (!batched) {
+      AWS_LOGSTREAM_ERROR(__func__, "Failed to record metric");
+    }
+
+    batched_count++;
+  }
+  return batched_count;
+}
```

Comment:
Is this change related to this PR? It seems to be changing the other metric message type? Could it be sent separately?

Reply:
I did not change anything for the other metric message type. The diff is being displayed sub-optimally.

Reply:
The diff looks better now.